src/components/Wizards/CreateVmWizard/tests/NetworksTab.test.js

```diff
@@ -1,13 +1,116 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { shallow, mount } from 'enzyme';
 import { NetworksTab } from '..';
 import NetworksTabFixture from '../fixtures/NetworksTab.fixture';
+import { TableFactory } from '../../../Table/TableFactory';
+import { POD_NETWORK } from '../../../../constants';
 
-const testNetworksTab = () => <NetworksTab {...NetworksTabFixture.props} />;
+const testNetworksTab = (props = NetworksTabFixture.props, mnt = false) => {
+  const networksTab = <NetworksTab {...props} />;
+  return mnt ? mount(networksTab) : shallow(networksTab);
+};
+
+const defaultNetworks = [
+  { id: 1, isBootable: false, name: 'nic1', mac: 'mac1', network: POD_NETWORK, errors: [] },
+  { id: 2, isBootable: true, name: 'nic2', mac: 'mac2', network: 'network2', errors: [] },
+  { id: 3, isBootable: false, name: 'nic3', mac: 'mac3', network: 'network3', errors: [] }
+];
 
 describe('<NetworksTab />', () => {
   it('renders correctly', () => {
-    const component = shallow(testNetworksTab());
+    const component = testNetworksTab();
     expect(component).toMatchSnapshot();
   });
+  it('passes existing networks to FormFactory', () => {
```

Comment:
this is implementation detail and IMO shouldn't be tested in unit tests

Reply:
Agree, it expects particular inner implementation.
Anyway, failing test here might point to a serious issue. How much complicated it would be to test the same "inner flow" as a black-box, so passing props and observing behavior as black-box? If feasible, this approach would bring additional value to us.